client/src/components/ContentNavigator/index.ts

```diff
@@ -247,6 +250,71 @@ class ContentNavigator implements SubscriptionProvider {
           "workbench.actions.treeView.contentdataprovider.collapseAll",
         );
       }),
+      commands.registerCommand(
+        "SAS.convertSasnbToFlw",
+        async (resource: ContentItem) => {
+          const resourceUri = getUri(resource);
+          // Ensure that the command is only available for .sasnb files
+          if (resourceUri.path.endsWith(".sasnb")) {
+            // Open window to chose the name and location of the new .flw file
+            const name = await window.showInputBox({
+              prompt: "Enter the name for the new .flw file",
+              value: resource.name.replace(".sasnb", ".flw"),
+            });
+
+            if (!name) {
+              // User canceled the input box
+              return;
+            }
+
+            if (!name.endsWith(".flw")) {
+              window.showErrorMessage(Messages.InvalidFlwFileNameError);
+              return;
+            }
+
+            // check that name does not exist
+            const parent = await this.contentDataProvider.getParent(resource);
+            const children = await this.contentDataProvider.getChildren(parent);
+            if (children.some((child) => child.name === name)) {
+              window.showErrorMessage(Messages.FileAlreadyExistsError);
+              return;
+            }
+
+            try {
+              const contentUri: string =
+                await this.contentDataProvider.provideTextDocumentContent(
+                  resourceUri,
+                );
+              const flowDataUint8Array = convert_sasnb_to_flw(contentUri, name);
+              // error if content is empty
+              if (flowDataUint8Array.length === 0) {
+                window.showErrorMessage(Messages.NoCodeToConvert);
+                return;
+              }
+              const newUri = await this.contentDataProvider.createFile(
+                parent,
+                name,
+              );
+              await this.contentDataProvider.writeFile(
+                newUri,
+                flowDataUint8Array,
+              );
```

Comment:
We should be able to call _only_ `createFile` with the flow data as the third parameter, I believe

Reply:
Yes indeed, done in 3ec847bd25ab2f905bbe0d735eac5ec89c4f6c03